Exploration-of-the-Oppia-stack.md

```diff
@@ -0,0 +1,431 @@
+## Table of Contents
+
+* [Support editing](#support-editing)
+  * [Frontend changes to support editing](#frontend-changes-to-support-editing)
+  * [Backend changes to support editing](#backend-changes-to-support-editing)
+* [Support viewing](#support-viewing)
+  * [Backend changes to support viewing](#backend-changes-to-support-viewing)
+  * [Frontend changes to support viewing](#frontend-changes-to-support-viewing)
+
+To better understand how Oppia works, let's work through a simple full-stack change--adding a field to a user's preferences. We'll discuss how to update the frontend code to display the new field and how to make the backend store this new field in the database.
+
+Before working through this tutorial, you should read through the [[Overview of Oppia's codebase|Overview-of-the-Oppia-codebase]].
```

Comment:
Is this going to be a `tutorial`? I think as per our last discussion we thought of this to be more of an explainer doc and less of a tutorial, right? (Sorry, if I had misunderstood, I'll be happy to have a quick discussion regarding the expectation of this PR on call, let me know if needed!)

Reply:
My understanding was that this page is a tutorial, but not a tutorial for how to write production code. Instead, it is a tutorial that walks readers through an exploration of the Oppia code.

Would "walk-through" as you suggested above work better?

Reply:
Based on my discussion with seanlip, this is actually more of a tutorial